File: 2022/03/31/distributeCandies.ts
/**
 * https://leetcode.com/problems/distribute-candies/
 * @author Andrew Jarombek
 * @since 3/31/2022
 */

function distributeCandies(candyType: number[]): number {
    let set = new Set<number>();

    for (let candy of candyType) {
        set.add(candy);
    }

    return Math.min(Math.floor(candyType.length / 2), set.size);
}

File: 2023/07/20/go/distribute_candies.go
/**
 * https://leetcode.com/problems/subtree-of-another-tree/
 * Author: Andrew Jarombek
 * Date: 7/20/2023
 */

package _go

import "math"

func distributeCandies(candyType []int) int {
	candyMap := make(map[int]struct{})

	for _, candy := range candyType {
		candyMap[candy] = struct{}{}
	}

	return int(math.Min(float64(len(candyMap)), float64(len(candyType)/2)))
}
